Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/RelatedTerm/GetAllByTermId/GetAllRelatedTermsByTermIdHandlerTest.cs

```diff
@@ -0,0 +1,121 @@
+
+            // Act
+            var result = await handler.Handle(query, CancellationToken.None);
+
+            // Assert
+            Assert.Equal("Cannot create DTOs for related words!", result.Errors.First().Message);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        public async Task Handle_ReturnsOkResult_WhenAllInputsAreValid(int id)
+        {
+            // Arrange
+            var query = new GetAllRelatedTermsByTermIdQuery(id);
+            var relatedTerms = new List<Entity> { new Entity() };
+            var relatedTermDTOs = new List<RelatedTermDTO> { new RelatedTermDTO() };
+            _mockRepository.Setup(x => x.RelatedTermRepository.GetAllAsync(It.IsAny<Expression<Func<Entity, bool>>>(),
+                It.IsAny<Func<IQueryable<Entity>, IIncludableQueryable<Entity, object>>>()))
+                .ReturnsAsync(relatedTerms);
+            _mockMapper.Setup(x => x.Map<IEnumerable<RelatedTermDTO>>(relatedTerms)).Returns(relatedTermDTOs);
+            var handler = new GetAllRelatedTermsByTermIdHandler(_mockMapper.Object, _mockRepository.Object);
+
+            // Act
+            var result = await handler.Handle(query, CancellationToken.None);
+
+            // Assert
+            Assert.True(result.IsSuccess);
+            Assert.Equal(relatedTermDTOs, result.Value);
+        }
+
+        [Theory]
+        [InlineData(1, 1, "RelatedTerm1")]
+        public async Task Handle_WhenRelatedTermsFound_ReturnsSuccessResult(int id, int termId, string testWord)
+        {
+            // Arrange
+            
+            var relatedTerm = new Entity { Id = id, TermId = termId, Word = testWord };
+            var relatedTerms = new List<Entity> { relatedTerm };
+            var relatedTermDTOs = new List<RelatedTermDTO> { new RelatedTermDTO() };
+            _mockMapper.Setup(x => x.Map<IEnumerable<RelatedTermDTO>>(relatedTerms)).Returns(relatedTermDTOs);
+
+            _mockRepository.Setup(x => x.RelatedTermRepository.GetAllAsync(It.IsAny<Expression<Func<Entity, bool>>>(),
+                It.IsAny<Func<IQueryable<Entity>, IIncludableQueryable<Entity, object>>>()))
+                .ReturnsAsync(relatedTerms);
+
+
+            var query = new GetAllRelatedTermsByTermIdQuery(termId);
+            var handler = new GetAllRelatedTermsByTermIdHandler(_mockMapper.Object, _mockRepository.Object);
+
+            // Act
+            var result = await handler.Handle(query, CancellationToken.None);
+
+            // Assert
+            Assert.True(result.IsSuccess);
+            Assert.IsAssignableFrom<IEnumerable<RelatedTermDTO>>(result.Value);
+            Assert.Equal(relatedTermDTOs.Count(), result.Value.Count());
+            Assert.Equal(relatedTermDTOs.First().Id, result.Value.First().Id);
+            Assert.Equal(relatedTermDTOs.First().TermId, result.Value.First().TermId);
+            Assert.Equal(relatedTermDTOs.First().Word, result.Value.First().Word);
+        }
```

Comment:
multiple assertions (look through existing test to learn how to avoid it)

Reply:
Fixed